include/Command.h

```diff
@@ -0,0 +1,158 @@
+#pragma once
+#include <string>
+#include <vector>
+#include <iostream>
+#include "constants.h"
+#include <fstream>
+#include <queue>
+#include <unistd.h>
+#include <sys/wait.h>
+#include <stdexcept>
+
+
+
+class Command {
+public:
+    virtual ~Command() = default;
+    virtual void execute() = 0;
+    /***
+    * Expands environment variables in a string to their values.
+    * @param str The string to expand.
+    * @return The expanded string.
+    */
+    std::string expandVariables(const std::string& str) {
+        std::string result;
+        std::size_t start = 0, end = 0;
+        while ((end = str.find_first_of("$}", start)) != std::string::npos) {
+            if (str[start] == '$' && end > start + 1) {
+                std::string varName = str.substr(start + 1, end - start - 1);
+                char* varValue = getenv(varName.c_str());
+                if (varValue != nullptr) {
+                    result += varValue;
+                }
+                start = end + 1;
+            }
+            else if (str[start] == '{' && end > start + 1 && str[end - 1] == '$') {
+                std::string varName = str.substr(start + 2, end - start - 2);
+                char* varValue = getenv(varName.c_str());
+                if (varValue != nullptr) {
+                    result += varValue;
+                }
+                start = end + 1;
+            }
+            else {
+                result += str.substr(start, end - start + 1);
+                start = end + 1;
+            }
+        }
+        if (start != str.length()) {
+            result += str.substr(start);
+        }
+        return result;
+    }
+
+};
+
+class EchoCommand : public Command {
```

Comment:
I think you have a misunderstanding, `echo` is a command like any other. It uses a compiled program called `echo` just like `grep`, `file` or any other command